nautobot/utilities/git.py

```diff
@@ -29,13 +33,12 @@ def checkout(self, branch, commit_hexsha=None):
         """
         Check out the given branch, and optionally the specified commit within that branch.
         """
-        # Short-circuit logic - do we already have this commit checked out?
-        if commit_hexsha == self.repo.head.commit.hexsha:
-            logger.debug(f"Commit {commit_hexsha} is already checked out.")
-            return commit_hexsha
-
         self.fetch()
         if commit_hexsha:
+            # Short-circuit logic - do we already have this commit checked out?
+            if commit_hexsha == self.repo.head.commit.hexsha:
+                logger.debug(f"Commit {commit_hexsha} is already checked out.")
+                return commit_hexsha
```

Comment:
Moving this down here means that we always run a `git fetch` even if the local clone already has the requested commit checked out. Is there any alternate approach we could take to avoid this unnecessary step?

Reply:
We could do `if commit_hexsha and commit_hexsha == self.repo.head.commit.hexsha:` I was trying to keep it simple, but I see the point in reducing commands run.